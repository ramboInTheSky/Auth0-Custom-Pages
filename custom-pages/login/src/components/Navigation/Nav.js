import React from 'react'
// import PropTypes from 'prop-types'
import { IndexLink, Link } from 'react-router'
import './Nav.scss'

export const Navigation = () => {
  const register = window.location.href.indexOf('register') !== -1
  return (
    <div className="navigation">
      {register ? (
        <div className="navigationLink">
          <h3> Create account </h3>
          <span>Sign in if you already have a NICE account.</span>
          <IndexLink
            data-qa-sel="Signin-link-login"
            to="/"
            activeClassName="activeRoute"
          >
            Sign in
          </IndexLink>
        </div>
      ) : (
        <div className="navigationLink">
          <h3> Log in </h3>
          <span>Need a NICE Account?</span>
          <Link
            data-qa-sel="Signup-link-login"
            to="/register"
            activeClassName="activeRoute"
          >
            Create an account
          </Link>
        </div>
      )}
    </div>
  )
}

// Navigation.propTypes = {
//   children: PropTypes.element.isRequired
// }

export default Navigation
