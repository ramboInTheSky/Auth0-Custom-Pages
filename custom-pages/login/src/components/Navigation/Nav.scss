@import '../variables.scss';

h3{
  font-weight: 700;
}

.activeRoute {
  font-weight: bold;
  text-decoration: none;
}

.navigation{
    // text-align: center;
    width: var(--col-width-mobile);
}

.navigationLink{
  font-weight: bold;
  font-size: 1rem;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.navigation span{
  padding-right: 5px;
}

.navigation a:hover{
    text-decoration: none;
}
