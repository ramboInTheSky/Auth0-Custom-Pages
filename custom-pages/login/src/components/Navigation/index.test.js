
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-14'

import Nav from './'

Enzyme.configure({ adapter: new Adapter() })

describe('Nav components', () => {
  it('should render <Nav /> correctly', () => {
    expect(shallow(<Nav />)).toMatchSnapshot()
  })
})
