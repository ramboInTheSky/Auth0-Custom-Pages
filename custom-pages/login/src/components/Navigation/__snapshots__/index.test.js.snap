// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav components should render <Nav /> correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="navigationLink"
      >
        <h3>
           Log in 
        </h3>
        <span>
          Need a NICE Account?
        </span>
        <Link
          activeClassName="activeRoute"
          data-qa-sel="Signup-link-login"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/register"
        >
          Create an account
        </Link>
      </div>,
      "className": "navigation",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
             Log in 
          </h3>,
          <span>
            Need a NICE Account?
          </span>,
          <Link
            activeClassName="activeRoute"
            data-qa-sel="Signup-link-login"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/register"
          >
            Create an account
          </Link>,
        ],
        "className": "navigationLink",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Log in ",
          },
          "ref": null,
          "rendered": " Log in ",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Need a NICE Account?",
          },
          "ref": null,
          "rendered": "Need a NICE Account?",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeClassName": "activeRoute",
            "children": "Create an account",
            "data-qa-sel": "Signup-link-login",
            "onlyActiveOnIndex": false,
            "style": Object {},
            "to": "/register",
          },
          "ref": null,
          "rendered": "Create an account",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="navigationLink"
        >
          <h3>
             Log in 
          </h3>
          <span>
            Need a NICE Account?
          </span>
          <Link
            activeClassName="activeRoute"
            data-qa-sel="Signup-link-login"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/register"
          >
            Create an account
          </Link>
        </div>,
        "className": "navigation",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
               Log in 
            </h3>,
            <span>
              Need a NICE Account?
            </span>,
            <Link
              activeClassName="activeRoute"
              data-qa-sel="Signup-link-login"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/register"
            >
              Create an account
            </Link>,
          ],
          "className": "navigationLink",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Log in ",
            },
            "ref": null,
            "rendered": " Log in ",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Need a NICE Account?",
            },
            "ref": null,
            "rendered": "Need a NICE Account?",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "activeRoute",
              "children": "Create an account",
              "data-qa-sel": "Signup-link-login",
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "/register",
            },
            "ref": null,
            "rendered": "Create an account",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFourteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
